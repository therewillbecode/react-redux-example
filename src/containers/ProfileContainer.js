import React, { Component } from "react";
import { connect } from "react-redux";

import Profile from "../components/Profile";
import { fetchProfile } from "../actions/index.js";

export class ProfileContainer extends Component {
  componentDidMount() {
    const { profile } = this.props;
    let token = localStorage.getItem("access_token");

    if (!profile && token) {
      this.props.fetchProfile(token);
    }
  }

  render() {
    const { profile } = this.props;

    return profile ? (
      <Profile profile={profile} />
    ) : (
      <img src="img/loading.svg" alt="loading" />
    );
  }
}

const mapStateToProps = state => ({
  profile: state.get("auth").get("profile")
});

const mapDispatchToProps = dispatch => {
  return {
    fetchProfile: token => dispatch(fetchProfile(token))
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(ProfileContainer);
