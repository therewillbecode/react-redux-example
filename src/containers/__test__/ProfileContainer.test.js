/* eslint-disable */
import React from "react";
import ReactShallowRenderer from "react-test-renderer/shallow";
import sinon from "sinon";
import { mount } from "enzyme";

import { ProfileContainer } from "../ProfileContainer";

class LocalStorageMock {
  constructor() {
    this.store = {};
  }

  getItem(key) {
    return this.store[key] || null;
  }

  setItem(key, value) {
    this.store[key] = value.toString();
  }

  removeItem(key) {
    delete this.store[key];
  }
}

const profile = {
  name: "Foo Bar",
  given_name: "Foo",
  family_name: "Bar",
  nickname: "Foobar",
  picture: "/photo.jpg"
};

describe("ProfileContainer", () => {
  it("renders loading icon if no profile data present", () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(<ProfileContainer />);
    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });

  it("renders profile if data present", () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(<ProfileContainer profile={profile} />);
    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });

  describe("fetchProfile", () => {
    let spy = null;

    beforeEach(() => {
      spy = new sinon.spy();
      global.localStorage = new LocalStorageMock();
      global.localStorage.setItem("access_token", "k");
    });

    it("is called in componentDidMount if there is no profile data", () => {
      const wrapper = mount(<ProfileContainer fetchProfile={spy} />);

      expect(spy.calledOnce).toBe(true);
    });

    it("is not called in componentDidMount if there is profile data", () => {
      const wrapper = mount(
        <ProfileContainer fetchProfile={spy} profile={profile} />
      );
      expect(spy.calledOnce).toBe(false);
    });
  });
});
