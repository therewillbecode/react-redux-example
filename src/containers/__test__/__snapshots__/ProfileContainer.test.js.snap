// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileContainer renders loading icon if no profile data present 1`] = `
<img
  alt="loading"
  src="img/loading.svg"
/>
`;

exports[`ProfileContainer renders profile if data present 1`] = `
<Profile
  profile={
    Object {
      "family_name": "Bar",
      "given_name": "Foo",
      "name": "Foo Bar",
      "nickname": "Foobar",
      "picture": "/photo.jpg",
    }
  }
/>
`;
